### PR TITLE
Improve error message of perform and add toString to all operations

### PR DESCRIPTION
**Pull request by cstamas**
CI
http://bamboo.s/browse/NX3-OSSF397

### CODE REVIEW

**Comment by jdillon** [comment]
FTR I'm still on the fence as to SimpleFormat or String.format.... more so with a single source file that has a mix of both.  Most toString() is only going to be used in debug calls that should defer rendering, so the minor speed improvement SimpleFormat provides over String.format is probably not worth it, and certainly not work using a mix of the 2.

File: plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java [hunk 1]
```diff
@@ -469,7 +475,7 @@ public Boolean execute(final StorageTx tx) {
 
       @Override
       public String toString() {
-        return "setLastVerified(" + path.getPath() + ", " + verified + ")";
+        return SimpleFormat.format("setLastVerified(%s)", path.getPath(), verified);
```
Comment (mcculls) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java]: missing second `%s`?

**Comment by mcculls** [comment]
+1 with comments

File: plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java (hunk 1 above)
Comment (jdillon) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java]: FTR most IDE understand String.format and likely point out missing tokens... where SimpleFormat.format() unless you train the IDE about non-standard replacement of String.format() will not.

Comment (cstamas) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java]: Yeah, I saw recent IDEA telling me about those. Will remove use of SimpleFormat

**Comment by cstamas** [comment]
PR updated, SimpleFormat use replaced with String.format, and message reworded to be more correct (attempt vs retry).

**Comment by cstamas** [comment]
4e2803a7aa9968cef69ec1700c133fb1bd20ef22

**Pull request closed without merging**